File: src/main/java/br/com/g1bet/service/AutenticacaoService.java
package br.com.g1bet.service;

import br.com.g1bet.model.Usuario;
import br.com.g1bet.repository.UsuarioRepository;
import br.com.g1bet.security.UserDetailsImpl;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class AutenticacaoService implements UserDetailsService {

    private final UsuarioRepository repository;

    public AutenticacaoService(UsuarioRepository repository) {
        this.repository = repository;
    }

    @Override
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        Optional<Usuario> usuario = repository.findByEmail(username);

        if (usuario.isPresent())
            return new UserDetailsImpl(usuario.get());
        throw new UsernameNotFoundException("Dados ínvalidos " + username);
    }

}

File: src/main/java/br/com/g1bet/service/AutenticacaoService.kt
package br.com.g1bet.service

import br.com.g1bet.repository.UsuarioRepository
import br.com.g1bet.security.UserDetailsImpl
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class AutenticacaoService(private val repository: UsuarioRepository) : UserDetailsService {
    @Throws(UsernameNotFoundException::class)
    override fun loadUserByUsername(username: String): UserDetails? {
        val usuario = repository.findByEmail(username)
        if (usuario!!.isPresent) return UserDetailsImpl(usuario.get())
        throw UsernameNotFoundException("Dados ínvalidos $username")
    }
}